src/Agent.Listener/MessageListener.cs

```diff
@@ -46,9 +45,9 @@ public sealed class MessageListener : AgentService, IMessageListener
             VssCredentials creds = credMgr.LoadCredentials();
             Uri uri = new Uri(serverUrl);
             VssConnection conn = ApiUtil.CreateConnection(uri, creds);
-            string sessionName = $"{Environment.MachineName}_{Guid.NewGuid().ToString()}";
-            var agentSystemCapabilities = new Dictionary<string, string>();
-            //TODO: add capabilities
+            string sessionName = $"{System.Environment.MachineName}_{Guid.NewGuid().ToString()}";
+            var enviroment = HostContext.GetService<IEnviroment>();
+            Dictionary<string, string> agentSystemCapabilities = await enviroment.GetCapabilities(token);
```

Comment:
remove any capabalilities with Null in the value. 'null' value of the capability may cause server problem.